code/modules/clothing/gloves/marine_gloves.dm

```diff
@@ -22,6 +22,42 @@
 	desc = "Insulated marine tactical gloves that protect against electrical shocks."
 	siemens_coefficient = 0
 
+/obj/item/clothing/gloves/marine/corpsman
+	name = "advanced medical combat gloves"
+	desc = "Advanced medical gloves, these include small electrodes to defibrilate a patiant. No more bulky units!"
+	icon_state = "defib_gloves"
+	item_state = "defib_gloves"
+	attack_verb = "Challanged ELECTRICALLY"
+	var/damage_threshold = 8 //This is the maximum non-oxy damage the defibrillator will heal to get a patient above -100, in all categories
+	var/charge_cost = 66 //How much energy is used.
+	var/obj/item/cell/dcell = null
+	var/datum/effect_system/spark_spread/sparks
+	var/defib_cooldown = 0 //Cooldown for toggling the defib
+
+/obj/item/clothing/gloves/marine/corpsman/Initialize()
+	. = ..()
+	dcell = new()
+	AddComponent(/datum/component/defiber, CALLBACK(src, .proc/usable, charge_cost), CALLBACK(src, .proc/get_cell))
+
+/obj/item/clothing/gloves/marine/corpsman/Destroy(force, ...)
+	. = ..()
+	var/datum/component/defiber/comp = GetComponent(/datum/component/defiber)
+	comp.RemoveComponent()
```

Comment:
does not clean up  sparks and dcell
```suggestion
/obj/item/clothing/gloves/marine/corpsman/Destroy(force, ...)
	. = ..()
```

Reply:
sparks are attached to the component, so it should be cleared up there